src/main/java/io/debezium/connector/yugabytedb/YugabyteDBOffsetContext.java

```diff
@@ -216,35 +203,38 @@ public SourceInfo getSourceInfo(YBPartition partition) {
 
     @Override
     public boolean isSnapshotRunning() {
-        return sourceInfo.isSnapshot();
+        // TODO: think about making this work
+        return true;
+        // return sourceInfo.isSnapshot();
     }
 
     @Override
     public void preSnapshotStart() {
-        sourceInfo.setSnapshot(SnapshotRecord.TRUE);
-        lastSnapshotRecord = false;
+        // Do nothing.
```

Comment:
If I throw `UnsupportedOperationException` here, then the base class which is calling these functions will all start hitting this.

We are only inheriting the methods but not using them to process our snapshot.